cronjobs/FileLoader.py

```diff
@@ -0,0 +1,128 @@
+import os, sys
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+
+from models import ScheduleHistory, CallingPointHistory
+from darwinpush.messagefactories.xml import ScheduleXMLMessageFactory
+import darwinpush.xb.pushport as pp
+
+import re
+
+class ScheduleFileLoader:
+
+    def __init__(self, file_location):
+        assert(file_location)
+        self.file_location = file_location
+
+    def update_daily_schedules(self):
+
+        collect_mode=False
+        journey_buffer = []
+
+        for line in open(self.file_location):
+
+            if '<Journey' in line:
+                assert(not collect_mode)
+                collect_mode=True
+
+            elif '</Journey' in line:
+                assert(collect_mode)
+
+                journey_buffer.append(line)
+                journey_string = "".join(journey_buffer)
```

Comment:
For instance, when you find `[…1…]</Journey>[…2…]`, you leave `[…2…]` for the next buffer and ignore it in the current buffer, in case it contains an opening tag for the next journey. 

This is pretty much what already happens but assumes there are newlines between them.

Anyway, don’t bother fixing if the files were OK so far. Need a less hacky solution in the long term anyway.

> On 3 Oct 2015, at 4:23 p.m., Alejandro Saucedo <notifications@github.com> wrote:
> 
> In cronjobs/FileLoader.py <https://github.com/HackPartners/bigdatadarwin/pull/16#discussion_r41088079>:
> 
> > +    def update_daily_schedules(self):
> > +
> > +        collect_mode=False
> > +        journey_buffer = []
> > +
> > +        for line in open(self.file_location):
> > +
> > +            if '<Journey' in line:
> > +                assert(not collect_mode)
> > +                collect_mode=True
> > +
> > +            elif '</Journey' in line:
> > +                assert(collect_mode)
> > +
> > +                journey_buffer.append(line)
> > +                journey_string = "".join(journey_buffer)
> I don't understand the solution proposed. So once we find the closing block, we would delete it from the string and re-consider the line?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/HackPartners/bigdatadarwin/pull/16/files#r41088079>.
>